packages/commerce-sdk-react/src/hooks/useDNT.ts

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import useAuthContext from './useAuthContext'
+import useConfig from './useConfig'
+
+interface useDntReturn {
+    dntStatus: boolean | undefined
+    updateDNT: (preference: boolean | null) => Promise<void>
+}
+
+/**
+ * Hook that returns
+ * dntStatus - a boolean indicating the current DNT preference
+ * updateDNT - a function that takes a DNT preference and creates the dw_dnt
+ *              cookie and reauthroizes with SLAS
+ *
+ * @group Helpers
+ * @category DNT
+ *
+ */
+const useDNT = (): useDntReturn => {
+    const auth = useAuthContext()
+    const config = useConfig()
+    const dwDntValue = auth.getDnt()
+
+    const updateDNT = async (preference: boolean | null) => {
+        let dntCookieVal = String(Number(preference))
```

Comment:
Can we move all of these logic inside auth.setDNT? Image we have another app that does not use React. It can reuse the logic in Auth Class. If we leave the logic here, it can only be used in React layer. The React Layer should be thin and let the Auth class handles the heavy duty

Reply:
@alexvuong is it possible for another non-React app to use the `Auth` class? I thought the `Auth` class is written in React 🤔

Reply:
@bfeister atm, we can't since we don't expose the Auth class, but technically if auth handles all the authentication and React layer is the layer that expose auth data and methods rather having those hooks handle business logic. It is easier if we have a case in the future that needs to reuse auth class in something that does not uses React.

Besides, It is also consistent with the pattern we are doing with utilities hooks like `useCustomerId` `useAccessToken`